### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by R4Ajeti**

### CODE REVIEW

File: Enumerable.rb [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    def my_any?
+        bool = false
+        self.my_each do |x|
+            if yield(x)
+                bool = true
+            end
+        end
+        bool
+    end
+    def my_none?
```
Comment (bolah2009) [inline Enumerable.rb]: #### `REQUIRED`

#### '#my_none?` should be able to take an argument.

> If instead a pattern is supplied, the method returns whether pattern === element for none of the collection members.

##### Learn more [here](https://ruby-doc.org/core-2.6.3/Enumerable.html#method-i-none-3F).

```ruby
$ irb
irb(main):001:0> require_relative './Enumerable'
=> true
irb(main):002:0> [1,2,3,4,5].none?(3)
=> false
irb(main):003:0> [1,2,3,4,5].none?(9)
=> true
irb(main):004:0> [1,2,3,4,5].my_none?(9)
Traceback (most recent call last):
        5: from /home/bola/.rbenv/versions/2.6.3/bin/irb:23:in `<main>'
        4: from /home/bola/.rbenv/versions/2.6.3/bin/irb:23:in `load'
        3: from /home/bola/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):4
        1: from /media/bola/.../code-reviews/Advanced_Building_Blocks/Enumerable.rb:41:in `my_none?'
ArgumentError (wrong number of arguments (given 1, expected 0))
irb(main):005:0> 
```

**Review by bolah2009 — CHANGES_REQUESTED** [review]
## Awesome Job :muscle: 

#### I tested all method only two seems to be inefficient. :smile: Nice. :wink: 

- See comments for required changes.
- Keep your code clean, please fix this. Currently `rubocop` reported **190 offenses**. :open_mouth:
You can fix this by running `rubocop -a` in the project root directory assuming you have `rubocop gem` installed.   
  > Keep your code [clean](https://github.com/github/rubocop-github/blob/master/STYLEGUIDE.md#ruby-style-guide).

Do this and submit for another code review request.

Good luck with your next submission. :star2: 

Cheers
[Bola](https://bolabuari.com/)

###### PS: Note that comments tagged as `OPTIONAL` are not part of the minimum requirements to approve this PR

File: Enumerable.rb [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+      my_each do |x|
+        count += 1 if args.first == x
+      end
+    else
+      raise ArgumentError
+    end
+    count
+  end
+
+  def my_count?
```
Comment (bolah2009) [inline Enumerable.rb]: :thinking: Is this the same with `def my_count(*args)` ?

**Review by bolah2009 — CHANGES_REQUESTED** [review]
## Hi @R4Ajeti 

The two methods **(`my_count` and `my_none`)** are still not working as expected as shown above., test the methods with the ruby methods. Visit the link attached in the previous comments to learn more about the behavior of these methods.

:bulb: Test the same method with different parameter types (eg block, numbers)

Do this and submit for another code review request.

Good luck with your next submission. :star2: 

Cheers
[Bola](https://bolabuari.com/)

**Review by jofer86 — CHANGES_REQUESTED** [review]
# :clap: :clap: Great Effort!!!!
Just a couple of minor fixes required on your code before you are able to move forward, please look at my comments below and DDT those bugs!!!.

Cheers, Jorge.

File: Enumerable.rb [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+      when 1
+        bool = false if x == args.first
+      else
+        raise ArgumentError
+      end
+    end
+    bool
+  end
+
+  def my_count(*args)
```
[the remark this answers is not in the document]
Comment (maelfosso) [reply Enumerable.rb]: # There is a new bug

Take a look below
```ruby
irb(main):004:0> [1,2,3,4,5,6,7,8].my_count
=> 8
irb(main):005:0> [1,2,3,4,5,6,7,8].count
=> 8
irb(main):006:0> [1,2,3,4,5,6,7,8].count(2)
=> 1
irb(main):007:0> [1,2,3,4,5,6,7,8].my_count(2)
=> 1
irb(main):008:0> [1,2,3,4,5,6,7,8].count { |x| x.even? }
=> 4
irb(main):009:0> [1,2,3,4,5,6,7,8].my_count { |x| x.even? }
=> 8
```

**Review by maelfosso — CHANGES_REQUESTED** [review]
Hello,

Really nice improvement.
But there is **just one bug to fix**
Check the comment below.

Kindly make the changes and re-submit a new code review request.
A thumb-up is waiting for you :-)

File: .rubocop.yml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  Max: 120
+
+# Too short methods lead to extraction of single-use methods, which can make
+# the code easier to read (by naming things), but can also clutter the class
+Metrics/MethodLength:
+  Include: ["app/controllers/*"]
+  Max: 20
+
+Metrics/BlockLength:
+  ExcludedMethods: ['describe']
```
Comment (bolah2009) [inline .rubocop.yml]: ### Please use the approprate [`.rubocop.yml` file](https://github.com/microverseinc/linters-config/blob/master/ruby/.rubocop.yml).
```yml
AllCops:
  Exclude:
    - "bin/*"
    - "db/*"
    - "config/*"
    - "Guardfile"
    - "Rakefile"
  DisplayCopNames: true

Metrics/LineLength:
  Max: 120
Metrics/MethodLength:
  Include: ["app/controllers/*"]
  Max: 20
Metrics/AbcSize:
  Include: ["app/controllers/*"]
  Max: 30
Metrics/ClassLength:
  Max: 150
Metrics/BlockLength:
  ExcludedMethods: ['describe']

Style/Documentation:
  Enabled: false
Style/ClassAndModuleChildren:
  Enabled: false

Layout/AlignHash:
  EnforcedColonStyle: key
Layout/ExtraSpacing:
  AllowForAlignment: false
Layout/MultilineMethodCallIndentation:
  Enabled: true
  EnforcedStyle: indented
```

**Review by bolah2009 — APPROVED** [review]
# Nice Job @R4Ajeti 

### Good Job implementing all suggested changes. It's a :+1: from me...

Use apprortaite `.rubocop.yml` file as shown in the comment below before merging, currently `rubucop` detected `64 offenses`. Kindly fix this too. :ok_hand: 

Good luck with your next project. :star2: 

Cheers
[Bola](https://bolabuari.com/)

**Pull request merged**